executor/inspection_result_test.go

```diff
@@ -368,12 +373,74 @@ func (s *inspectionResultSuite) TestThresholdCheckInspection2(c *C) {
 		"scheduler-cmd-duration tikv tikv-0 5.00 < 0.10 select instance, max(value) as max_value from metrics_schema.tikv_scheduler_command_duration where time>='2020-02-12 10:35:00' and time<='2020-02-12 10:37:00' and quantile=0.99 group by instance having max_value > 0.100000;",
 		"scheduler-pending-cmd-count tikv tikv-0 1001.00 < 1000.00 select instance, max(value) as max_value from metrics_schema.tikv_scheduler_pending_commands where time>='2020-02-12 10:35:00' and time<='2020-02-12 10:37:00' group by instance having max_value > 1000.000000;",
-		"storage-snapshot-duration tikv tikv-0 0.20 < 0.05 select instance, max(value) as max_value from metrics_schema.tikv_storage_async_request_duration where time>='2020-02-12 10:35:00' and time<='2020-02-12 10:37:00' and type='snapshot' group by instance having max_value > 0.050000;",
+		"storage-snapshot-duration tikv tikv-0 0.06 < 0.05 select instance, max(value) as max_value from metrics_schema.tikv_storage_async_request_duration where time>='2020-02-12 10:35:00' and time<='2020-02-12 10:37:00' and type='snapshot' group by instance having max_value > 0.050000;",
 		"storage-write-duration tikv tikv-0 0.20 < 0.10 select instance, max(value) as max_value from metrics_schema.tikv_storage_async_request_duration where time>='2020-02-12 10:35:00' and time<='2020-02-12 10:37:00' and type='write' group by instance having max_value > 0.100000;",
 		"tso-duration tidb pd-0 0.06 < 0.05 select instance, max(value) as max_value from metrics_schema.pd_tso_wait_duration where time>='2020-02-12 10:35:00' and time<='2020-02-12 10:37:00' and quantile=0.999 group by instance having max_value > 0.050000;",
 	))
 }
 
+func (s *inspectionResultSuite) TestThresholdCheckInspection3(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	// Mock for metric table data.
+	fpName := "github.com/pingcap/tidb/executor/mockMetricsTableData"
+	c.Assert(failpoint.Enable(fpName, "return"), IsNil)
+	defer func() { c.Assert(failpoint.Disable(fpName), IsNil) }()
+
+	datetime := func(s string) types.Time {
+		t, err := types.ParseTime(tk.Se.GetSessionVars().StmtCtx, s, mysql.TypeDatetime, types.MaxFsp)
+		c.Assert(err, IsNil)
+		return t
+	}
+
+	// construct some mock abnormal data
+	mockData := map[string][][]types.Datum{
+		"tikv_thread_cpu":                     {},
+		"pd_tso_wait_duration":                {},
+		"tidb_get_token_duration":             {},
+		"tidb_load_schema_duration":           {},
+		"tikv_scheduler_command_duration":     {},
+		"tikv_handle_snapshot_duration":       {},
+		"tikv_storage_async_request_duration": {},
+		"tikv_engine_write_duration":          {},
+		"tikv_engine_max_get_duration":        {},
+		"tikv_engine_max_seek_duration":       {},
+		"tikv_scheduler_pending_commands":     {},
+		"tikv_block_index_cache_hit":          {},
+		"tikv_block_data_cache_hit":           {},
+		"tikv_block_filter_cache_hit":         {},
+		"pd_scheduler_store_status": {
```

Comment:
You don't mock this data and just rewrite your SQL with an addtional condition `where item in ('leader_score', '...')`

Reply:
great, done.